Index: service/service/java/source/etc/workspace-service/global-policies.conf
===================================================================
RCS file: /home/globdev/CVS/globus-packages/workspace/vm/service/service/java/source/etc/workspace-service/global-policies.conf,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 global-policies.conf
--- service/service/java/source/etc/workspace-service/global-policies.conf	8 Sep 2008 19:50:11 -0000	1.1.1.1
+++ service/service/java/source/etc/workspace-service/global-policies.conf	7 Jul 2009 16:52:09 -0000
@@ -67,3 +67,11 @@ termination.offset.seconds=600
 # per-use if one justs wants to limit all static IP use
 
 allow.static.addresses=false
+
+
+# POLICY: allowed.http.hosts
+#
+# This is a list of hosts that the users are allowed to propagate images from 
+# via http propagation. This should be a comma separated list. 
+
+allowed.http.hosts=
Index: service/service/java/source/etc/workspace-service/other/main.xml
===================================================================
RCS file: /home/globdev/CVS/globus-packages/workspace/vm/service/service/java/source/etc/workspace-service/other/main.xml,v
retrieving revision 1.2
diff -u -p -r1.2 main.xml
--- service/service/java/source/etc/workspace-service/other/main.xml	9 Jan 2009 20:07:20 -0000	1.2
+++ service/service/java/source/etc/workspace-service/other/main.xml	7 Jul 2009 16:52:09 -0000
@@ -143,6 +143,8 @@
                   value="$GLOBAL{termination.offset.seconds}" />
         <property name="defaultRunningTimeSeconds"
                   value="$GLOBAL{default.runtime.seconds}" />
+        <property name="allowedHttpHosts"
+                  value="$GLOBAL{allowed.http.hosts}" />
 
         <!-- Property values coming via common.conf -->
         <property name="fake" value="$COMMON{fake.mode}" />
Index: service/service/java/source/src/org/globus/workspace/service/binding/GlobalPolicies.java
===================================================================
RCS file: /home/globdev/CVS/globus-packages/workspace/vm/service/service/java/source/src/org/globus/workspace/service/binding/GlobalPolicies.java,v
retrieving revision 1.1
diff -u -p -r1.1 GlobalPolicies.java
--- service/service/java/source/src/org/globus/workspace/service/binding/GlobalPolicies.java	8 Sep 2008 20:06:15 -0000	1.1
+++ service/service/java/source/src/org/globus/workspace/service/binding/GlobalPolicies.java	7 Jul 2009 16:52:09 -0000
@@ -28,6 +28,8 @@ public interface GlobalPolicies {
 
     public String[] getVmmVersions();
 
+    public String[] getAllowedHttpHosts();
+
     public String getVmm();
 
     public int getMaximumGroupSize();
Index: service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultBindDisks.java
===================================================================
RCS file: /home/globdev/CVS/globus-packages/workspace/vm/service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultBindDisks.java,v
retrieving revision 1.1
diff -u -p -r1.1 DefaultBindDisks.java
--- service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultBindDisks.java	8 Sep 2008 20:06:16 -0000	1.1
+++ service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultBindDisks.java	7 Jul 2009 16:52:09 -0000
@@ -140,6 +140,7 @@ public class DefaultBindDisks implements
 
         final String rootDiskScheme = uri.getScheme();
         final boolean local = "file".equals(rootDiskScheme);
+        final boolean http = "http".equals(rootDiskScheme);
 
         if (!propagationEnabled && !local) {
             final String err = "cannot propagate: supplied image '" +
@@ -148,6 +149,23 @@ public class DefaultBindDisks implements
             throw new CreationException(err);
         }
 
+        if (propagationEnabled && http) {
+            boolean foundHost = false;
+
+	    String[] hosts = this.globals.getAllowedHttpHosts();
+            for (String host : hosts) {
+                if (uri.getHost().equals(host)) {
+                    foundHost = true;
+                }
+	    }
+
+            if (!foundHost) {
+                final String err = "cannot propagate: supplied image '" +
+                        uri.toASCIIString() + "' is not in allowed.http.hosts";
+                throw new CreationException(err);
+            }
+        }
+
         final VirtualMachinePartition partition = new VirtualMachinePartition();
 
         if (propagationEnabled && !local) {
Index: service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultGlobalPolicies.java
===================================================================
RCS file: /home/globdev/CVS/globus-packages/workspace/vm/service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultGlobalPolicies.java,v
retrieving revision 1.1
diff -u -p -r1.1 DefaultGlobalPolicies.java
--- service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultGlobalPolicies.java	8 Sep 2008 20:06:16 -0000	1.1
+++ service/service/java/source/src/org/globus/workspace/service/binding/defaults/DefaultGlobalPolicies.java	7 Jul 2009 16:52:09 -0000
@@ -52,6 +52,8 @@ public class DefaultGlobalPolicies imple
     private boolean fake;
     private long fakelag;
 
+    private String[] allowedHttpHosts;
+
 
     // -------------------------------------------------------------------------
     // GET/SET
@@ -97,6 +99,14 @@ public class DefaultGlobalPolicies imple
         this.vmmVersions = vmmVersions;
     }
 
+    public String[] getAllowedHttpHosts() {
+        return this.allowedHttpHosts;
+    }
+
+    public void setAllowedHttpHosts(String[] allowedHttpHosts) {
+        this.allowedHttpHosts = allowedHttpHosts;
+    }
+
     public String getVmm() {
         return this.vmm;
     }
